ape_hardhat/providers.py

```diff
@@ -382,9 +382,25 @@ def get_transaction_trace(self, txn_hash: str) -> Iterator[TraceFrame]:
         Returns:
             Iterator(TraceFrame): Transaction execution trace object.
         """
-        logs = self._make_request("debug_traceTransaction", [txn_hash]).structLogs
-        for log in logs:
-            yield TraceFrame(**log)
+
+        for frame in self._make_request("debug_traceTransaction", [txn_hash]).structLogs:
+            yield TraceFrame(**frame)
+
+    def get_calltree_from_trace(
+        self, txn_hash: str, show_internals=False, **root_node_kwargs
+    ) -> CallTreeNode:
+        txn = self.get_transaction(txn_hash=txn_hash)
+        root_node_kwargs["gas_cost"] = txn.gas_used
```

Comment:
Is there a chance the user passes in some of these kwargs? If that happens, they will get overridden by the `txn` properties.

Do we need to do:

```python
root_node_kwargs["gas_cost"] = root_node_kwargs.get("gas_cost") or txn.gas_used
...
```

I am legitimately asking, as I am not sure!